src/store/wallet/wallet.hooks.ts

```diff
@@ -0,0 +1,138 @@
+import { useAtomCallback, useAtomValue, useUpdateAtom } from 'jotai/utils';
+import type { InMemoryVault } from '@background/vault';
+import type { VaultActions } from '@background/vault-types';
+import { gaiaUrl } from '@common/constants';
+import { useOnboardingState } from '@common/hooks/auth/use-onboarding-state';
+import { textToBytes } from '@common/store-utils';
+import { finalizeAuthResponse } from '@common/utils';
+import {
+  createWalletGaiaConfig,
+  getOrCreateWalletConfig,
+  makeAuthResponse,
+  updateWalletConfigWithApp,
+} from '@stacks/wallet-sdk';
+import { currentAccountIndexState, currentAccountStxAddressState } from '@store/accounts';
+import { localNonceState } from '@store/accounts/nonce';
+import { currentNetworkState } from '@store/network/networks';
+import { useAtom } from 'jotai';
+import { useCallback } from 'react';
+import {
+  encryptedSecretKeyStore,
+  hasRehydratedVaultStore,
+  hasSetPasswordState,
+  lastSeenStore,
+  secretKeyState,
+  walletState,
+} from './wallet';
+
+export function useHasRehydratedVault() {
+  return useAtomValue(hasRehydratedVaultStore);
+}
+
+export function useWalletState() {
+  return useAtom(walletState);
+}
+
+export function useSecretKey() {
+  return useAtomValue(secretKeyState);
+}
+
+export function useEncryptedSecretKeyStore() {
+  return useAtomValue(encryptedSecretKeyStore);
+}
+
+export function useHasSetPasswordState() {
+  return useAtomValue(hasSetPasswordState);
+}
+
+export function useUpdateLastSeenStore() {
+  return useUpdateAtom(lastSeenStore);
+}
+
+export function useSetLatestNonce() {
+  return useAtomCallback<void, number>(
+    useCallback((get, set, newNonce) => {
+      if (newNonce !== undefined) {
+        const network = get(currentNetworkState);
+        const address = get(currentAccountStxAddressState);
+        if (!address) return;
+        // we increment here for next nonce
+        set(localNonceState([address, network.url]), newNonce + 1);
+      }
+    }, [])
+  );
+}
+
+export function useFinishSignIn() {
```

Comment:
probably for a future improvement, but I personally like the convention of adding `Callback` to the end of hooks that only return a method, so this would be `useFinishSignInCallback`, thoughts?

Reply:
sure